snippets/pyaloha/ccode.py

```diff
@@ -28,33 +28,43 @@ class CEVENTTIME(ctypes.Structure):
         ('server_upload', ctypes.c_uint64)
     ]
 
-    __slots__ = (
-        'client_creation', 'server_upload',
-        'dtime', 'is_accurate'
-    )
-
     delta_past = datetime.timedelta(days=6 * 30)
     delta_future = datetime.timedelta(days=1)
 
-    def _setup_time(self):
+    def get_approx_time(self):
+        if not self.dtime:
+            self._setup_time()
+        return self.dtime, self.is_accurate
+
+    def make_object(self):
         client_dtime = SerializableDatetime.utcfromtimestamp(
             self.client_creation / 1000.  # timestamp is in millisecs
         )
         server_dtime = SerializableDatetime.utcfromtimestamp(
             self.server_upload / 1000.  # timestamp is in millisecs
         )
 
-        self.dtime = server_dtime
-        self.is_accurate = False
+        dtime = server_dtime
         if client_dtime >= server_dtime - CEVENTTIME.delta_past and\
                 client_dtime <= server_dtime + CEVENTTIME.delta_future:
-            self.dtime = client_dtime
-            self.is_accurate = True
+            dtime = client_dtime
 
-    def get_approx_time(self):
-        if not self.dtime:
-            self._setup_time()
-        return self.dtime, self.is_accurate
+        pytime = PythonEventTime()
+        pytime.client_creation = client_dtime
+        pytime.server_upload = server_dtime
+        pytime.dtime = dtime
+        return pytime
+
+
+class PythonEventTime(object):
+    __slots__ = (
+        'client_creation', 'server_upload',
+        'dtime', 'is_accurate'
```

Comment:
python3.5:
``ValueError: 'is_accurate' in _slots_ conflicts with class variable``

Reply:
done